%3a1/srfi-1-reference.scm

```diff
@@ -713,18 +753,18 @@
 ;;; the needs of the fold/map procs -- for example, to minimize the number
 ;;; of times the argument lists need to be examined.
 
-;;; Return (map cdr lists). 
+;;; Return (map cdr lists).
 ;;; However, if any element of LISTS is empty, just abort and return '().
 (define (%cdrs lists)
-  (let f ((ls lists) (d* '()))
+  (let f ((ls lists))
     (if (pair? ls)
-      (let ((x (car ls)))
-        (if (null-list? x)
-          '()
-          (f (cdr ls) (cons (cdr x) d*))))
-      (reverse d*))))
+        (let ((x (car ls)))
+          (if (null? x)
+              '()
+              (cond (cdr x) (f (cdr ls)))))
```

Comment:
Should this really be `cond` here?

Reply:
Oops! Don't know how tests didn't get this! Maybe we need more test cases.